packages/devtools_app/lib/src/memory/memory_tracker_model.dart

```diff
@@ -160,78 +151,142 @@ class TreeTracker {
+          final allocation = TrackerAllocation(
+            stacktrace.timestamp,
+            stacktrace,
+          );
           classEntry.addChild(allocation);
-          int displayDepth = 0;
-          for (var callStackEntry in stacktrace.stacktrace) {
-            if (allocation.displayAll ||
-                displayDepth++ < allocation.callDepthDisplay) {
-              allocation.addChild(TrackerCall(callStackEntry));
-            }
-          }
-          if (!allocation.displayAll) {
-            allocation.addChild(TrackerMore(stacktrace.stacktrace));
-          }
         }
       }
     }
   }
 
-  /// Remove the "More ..." and show the entire call stack.
-  void expandCallStack(TrackerMore item) {
-    final TrackerAllocation allocation = item.parent;
-
-    // Show the entire call stack for this instance.
-    allocation.displayAll = true;
-
-    // Re-construct the callstack for this instance.
-    final leafNodesCount = allocation.children.length;
-
-    assert(allocation.children.last is TrackerMore);
-    allocation.removeLastChild(); // Remove "More ..."
+  Widget _helpTrackScreen(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const Text('To track allocations for a class, enable the '
+            'checkbox for that class in the table above.'),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Text('After interacting with your app, come '
+                'back to this tool and click the track button '),
+            trackImage(context),
+          ],
+        ),
+        const Text('to view the collected stack '
+            'traces of constructor calls.'),
+      ],
+    );
+  }
 
-    // Add the entire call stack as leaf nodes.
-    final totalCallStack = item.stacktrace.length;
-    for (var index = leafNodesCount - 1; index < totalCallStack; index++) {
-      allocation.addChild(TrackerCall(item.stacktrace[index]));
-    }
-    allocation.expandCascading();
+  Widget _helpStacktraceScreen(BuildContext context) => Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: const [
+          Text('Select an instance of a tracked class to'
+              ' view its allocation stack trace.'),
+        ],
+      );
+
+  Widget displaySelectedStackTrace(
+    BuildContext context,
+    MemoryController controller,
+    ScrollController scroller,
+    TrackerAllocation tracker,
+  ) {
+    final allocationStacktrace = tracker.stacktrace;
+
+    final stackTrace = allocationStacktrace.stacktrace;
+    final sources = allocationStacktrace.sources;
+    final callstackLength = stackTrace.length;
+    assert(sources.length == callstackLength);
+
+    final titleBackground = titleSolidBackgroundColor(Theme.of(context));
+
+    return Column(
+      children: [
+        SizedBox(
+          width: double.infinity,
+          height: defaultRowHeight + 4, // From table.dart
```

Comment:
see high level comment about too much spacing.

Reply:
This is the title bar to match the Table's header height.  The contents of the list have been compressed by eliminating ListTile.